File: app/src/main/java/com/neome/java/api/meta/base/dto/EntVdEventRpcCall.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.EnumDefnHttpMethod;
import com.neome.java.api.meta.base.Types.KeychainId;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class EntVdEventRpcCall extends EntVdEvent
{
  @Nullable
  public KeychainId[] allowedKeychainIdSet;

  @Nullable
  public EnumDefnHttpMethod apiMethod;

  @Nullable
  public String apiName;

  @Nullable
  public Boolean executeAsync;

  @Nullable
  public FormRefKey inputForm;

  @Nullable
  public FormRefKey outputForm;

  @Nullable
  public StudioPluginApiBody requestBody;

  @Nullable
  public StudioPluginApiBody responseBody;
}

File: app/src/main/java/com/neome/api/meta/base/dto/EntVdEventRpcCall.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnHttpMethod
import com.neome.api.meta.base.Types.KeychainId

interface EntVdEventRpcCall : EntVdEvent {
    val allowedKeychainIdSet: List<KeychainId>?
    val apiMethod: EnumDefnHttpMethod?
    val apiName: String?
    val executeAsync: Boolean?
    val inputForm: FormRefKey?
    val outputForm: FormRefKey?
    val requestBody: StudioPluginApiBody?
    val responseBody: StudioPluginApiBody?
}
